G/GDAL/build_tarballs.jl

```diff
@@ -3,27 +3,28 @@
 using BinaryBuilder, Pkg
 
 name = "GDAL"
-upstream_version = v"3.6.2"
-version_offset = v"1.0.0"
+upstream_version = v"3.7.1"
```

Comment:
3.7.2 is out now, let's go straight to that: https://github.com/OSGeo/gdal/blob/v3.7.2/NEWS.md